### DISCUSSION TITLE
Custom CSS-IDs and classes for categories and articles

### DISCUSSION THREAD

**Discussion by itnomic** [PrestaShop/PrestaShop#41063] [Ideas]
#### Is your feature request related to a problem?

In PrestaShop each page has its own ID or CSS class and can be designed individually, but it is not possible to combine several pages with one CSS or ID class.

#### Describe the solution you'd like

I think it would be helpful if you have the opportunity to add your own CSS classes and IDs to article and maybe also on category pages (Screenshot is only an example, in this case you can add the CSS rules to the article options).

#### Additional context

![customclass](https://user-images.githubusercontent.com/5217081/128312138-789c3f6c-0059-4d7f-bea5-a7fbb443d5e0.jpg)

**Comment by Hlavtox** [comment]
@itnomic `page-category tax-display-enabled category-id-3 category-clothes category-id-parent-2 category-depth-level-2` You have all these classes on `<body>` tag, is it not enough? :-)

**Comment by itnomic** [comment]
Hey @Hlavtox, I give you an example: 

I have one category with 100 products and would like to change the page-design of 30 products.

In css I have to write

```
.product-id-1 h1, 
.product-id-5 h1,
.product-id-28 h1,
.product-id-55 h1, 
.product-id-... h1 {
    font-size: 2em;
}
```
Whenever I have a **new** product with the other design, I have to edit my stylesheet every time again, thats not effective and not the best way for editors. 

If I have my own class i can write for all product-pages that have to be changed:

```
.myownclass h1 {
   font-size: 2em; 
}

```
If an editor wants to have the h1-tag with 2em, he only needs to complete the class "myownclass" in the backoffice, without editing any stylesheets.

**Comment by hibatallahAouadni** [comment]
Hello @itnomic 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

PS: thanks @Hlavtox for your quick reply :wink: 

Thank you
